From 2a22cdc0a532a0d0aef8cef4ea6ca559c0f5c51c Mon Sep 17 00:00:00 2001
From: jschyra <jschyra@90b65887-3827-0410-9a23-83215b262276>
Date: Tue, 7 Jun 2016 14:51:59 +0000
Subject: [PATCH 0683/1000] MANTIS 0041253 [SetKioskData]:  - Add verbose
 output to joinDom.cmd  - Update README.md  - Update status messages

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1032 90b65887-3827-0410-9a23-83215b262276
---
 .../Build/setup/SetKioskData/README.md         |  1 +
 .../Build/setup/SetKioskData/SetKioskData.iss  | 18 ++++++++++--------
 .../Build/setup/SetKioskData/joinDom.cmd       |  4 ++--
 3 files changed, 13 insertions(+), 10 deletions(-)

diff --git a/SelfServiceCommon/Build/setup/SetKioskData/README.md b/SelfServiceCommon/Build/setup/SetKioskData/README.md
index 1b3eea05..b6bf4555 100644
--- a/SelfServiceCommon/Build/setup/SetKioskData/README.md
+++ b/SelfServiceCommon/Build/setup/SetKioskData/README.md
@@ -13,6 +13,7 @@ SetKioskData have the following prerequironments:
 ### Install PowerShell 4
 
 To install PowerShell 4 as a part of the Windows Management Framework 4.0, you have to run Windows6.1-KB2819745-x86-MultiPkg.msu setup.
+The Windows Management Framework 4.0 needs .NET 4.x. If you don't have .NET 4.0 installed, you can install it Manually with NDP452-KB2901907-x86-x64-AllOS-ENU.exe
 After updating and rebooting the system, you can use the new powershell. 
 To test which powershell version is installed, please use following command: `powershell $PSVersionTable.PSVersion`
 
diff --git a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
index 4f8b56da..3a5e9db7 100644
--- a/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
+++ b/SelfServiceCommon/Build/setup/SetKioskData/SetKioskData.iss
@@ -36,16 +36,16 @@ SetupIconFile=massai30.ico
 AlwaysRestart=True
 
 [Files]
-Source: "configNetwork.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigNetwork()
-Source: "configHostname.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigHostname(); Check: RunSetHostname()
-Source: "joinDom.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunJoinDomain(); Check: RunAddDomain()
-Source: "configEnvironment.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigEnvironment(); 
-Source: "configConveyor.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigConveyor(); Check: RunConveyor()
-Source: "configApplication.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigApplication()
-Source: "{code:GetMIPSUserDir}\Start MIPS.lnk"; DestDir: "{userstartup}"; Flags: external skipifsourcedoesntexist
+Source: "configNetwork.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigNetwork();
+Source: "configHostname.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigHostname(); Check: RunSetHostname();
+Source: "joinDom.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunJoinDomain(); Check: RunAddDomain();
+Source: "configEnvironment.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigEnvironment();
+Source: "configConveyor.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigConveyor(); Check: RunConveyor();
+Source: "configApplication.cmd"; DestDir: "{tmp}"; Flags: deleteafterinstall; AfterInstall: RunConfigApplication();
+Source: "{code:GetMIPSUserDir}\Start MIPS.lnk"; DestDir: "{userstartup}"; Flags: external skipifsourcedoesntexist;
 
 [Messages]
-ButtonInstall=&Update
+ButtonInstall=&Localize
 WelcomeLabel2=This will localice the identifier and hostname on your kiosk.
 ReadyLabel1=Setup is now ready to begin localizing the identifier and hostname on your kiosk.
 ExitSetupMessage=Setup is not complete. If you quit now, the kiosk will not be updated.%n%nYou may run the Setup program again at another time to complete the update.%n%nExit Setup?
@@ -53,6 +53,8 @@ ReadyLabel2b=Click Update to continue with the update.
 WizardReady=Ready to update
 FinishedLabel=Setup has finished updating the identifier and hostname on your computer.%nPlease update the IP address manually and then reboot the kiosk to finish the kiosk localisation.
 ReadyLabel2a=Click Update to continue with the update, or click Back if you want to review or change any settings.
+StatusExtractFiles=Localizing image...
+StatusRollback=Aborting Setup...
 
 [InstallDelete]
 Name: {userstartup}\SetKioskData.lnk; Type: files
diff --git a/SelfServiceCommon/Build/setup/SetKioskData/joinDom.cmd b/SelfServiceCommon/Build/setup/SetKioskData/joinDom.cmd
index 8fd62336..eb61dd37 100644
--- a/SelfServiceCommon/Build/setup/SetKioskData/joinDom.cmd
+++ b/SelfServiceCommon/Build/setup/SetKioskData/joinDom.cmd
@@ -47,8 +47,8 @@ if /I "%NAME%" == "%COMPUTERNAME%" (
 )
 
 rem "%PS%" "Add-Computer -DomainName %DOMAIN% -ComputerName %COMPUTERNAME% -NewName %NAME% -Credential %USER% %OU_PATH%"
-echo Run PS Command: "%PS%" "Add-Computer -DomainName %DOMAIN% %NEW_NAME% -Credential %USER% %OU_PATH%" >> "%LOG%"
-"%PS%" "Add-Computer -DomainName %DOMAIN% %NEW_NAME% -Credential %USER% %OU_PATH%" >> "%LOG%" 2>&1
+echo Run PS Command: "%PS%" "Add-Computer -DomainName %DOMAIN% %NEW_NAME% -Credential %USER% %OU_PATH% -verbose -passthru" >> "%LOG%"
+"%PS%" "Add-Computer -DomainName %DOMAIN% %NEW_NAME% -Credential %USER% %OU_PATH% -verbose -passthru" >> "%LOG%" 2>&1
 
 if not "%errorlevel%" == "0" (
     echo PS-Command failed! Abort! >> "%LOG%"
-- 
2.41.0.windows.1

